src/ui/components/shared/JumpToCodeButton.tsx

```diff
@@ -0,0 +1,87 @@
+import { ExecutionPoint } from "@replayio/protocol";
+import classnames from "classnames";
+import React, { useState } from "react";
+
+import Icon from "replay-next/components/Icon";
+import { isExecutionPointsGreaterThan } from "replay-next/src/utils/time";
+
+export type JumpToCodeFailureReason = "not_loaded" | "no_hits";
+export type JumpToCodeStatus = JumpToCodeFailureReason | "not_checked" | "loading" | "found";
+
+export const errorMessages: Record<JumpToCodeFailureReason, string> = {
+  not_loaded: "Not loaded",
+  no_hits: "No results",
+};
+
+interface JumpToCodeButtonProps {
+  status: JumpToCodeStatus;
+  onClick: () => void;
+  currentExecutionPoint: ExecutionPoint | null;
+  targetExecutionPoint: ExecutionPoint;
+}
+
+export function JumpToCodeButton({
+  status,
+  onClick,
+  currentExecutionPoint,
+  targetExecutionPoint,
+}: JumpToCodeButtonProps) {
+  const [isHovered, setIsHovered] = useState(false);
```

Comment:
Since this variable only impacts styles, it could probably be done more cleanly using CSS `:hover` state (and avoid the additional re-render)

Reply:
That hovering behavior has gone through a couple copy-paste cycles at this point. I _think_ it was there in the console message items where we got the general behavior from in the first place? Agreed it could probably be simplified.